### DISCUSSION TITLE
Python is broken after last streamlink update

### DISCUSSION THREAD

**Discussion by campones** [streamlink/streamlink#5494] [General]
### Description

Hello
Updated today to the last streamlink release, I then met a lot of issues .  Currently running streamlink 6.0.1

it broke my python somehow

"
ERROR: requests 2.27.0 has requirement urllib3<1.27,>=1.21.1, but you'll have urllib3 2.0.4 which is incompatible."

"streamlink
/usr/local/lib/python3.8/dist-packages/requests/__init__.py:102: RequestsDependencyWarning: urllib3 (2.0.4) or chardet (3.0.4)/charset_normalizer (2.0.9) doesn't match a supported version!
  warnings.warn("urllib3 ({}) or chardet ({})/charset_normalizer ({}) doesn't match a supported "

> pip3 install --upgrade requests

after upgrading requests, the error isn't showing anymore, however my certbot was still broken.  and I need it more than streamlink these days (not using currently streamlink in this server)

I upgraded certbot and sounded fine but after a reboot I got a problem with pyopenssl 

sudo rm -rf /usr/lib/python3/dist-packages/OpenSSL
sudo pip3 install pyopenssl
sudo pip3 install pyopenssl --upgrade

I haven't reboot . hopefully all will be fine.

**Comment by mkbloke** [comment]
I don't know what distribution you're using, but system Python packages will usually be managed via a package manager.  Trampling over them with `pip` is usually a very bad idea.  If you want to use `pip`, install into a [virtual environment](https://streamlink.github.io/install.html#virtual-environment) to isolate Streamlink's dependencies from those of the system and other Python applications.  You could also consider using the [AppImage](https://streamlink.github.io/install.html#linux-appimages).

**Comment by campones** [comment]
yes, I should have specified of course.  OS  Ubuntu 20.04 lts 64bits 
never had an issue after upgrading streamlink till today 

I have always proceeded like this in the past, I guess I have been lucky
